src/Microsoft.AspNetCore.Server.Kestrel/Internal/Infrastructure/HttpUtilities.cs

```diff
@@ -60,6 +61,33 @@ static HttpUtilities()
             _methodNames[(byte)HttpMethod.Options] = HttpMethods.Options;
         }
 
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static int GetKnownMethodIndex(ulong value)
+        {
+            var tmp = (int)value & 0x100604;
+
+            return ((tmp >> 2) | (tmp >> 8) | (tmp >> 17)) & 0x0F;
```

Comment:
Perfect hash? Did you write a program to generate the constants in this method? If so can you post it in the PR comments?